src/Core/src/Platform/Windows/ShadowExtensions.cs

```diff
@@ -24,28 +23,33 @@ public static async Task<CompositionBrush> GetAlphaMaskAsync(this UIElement elem
 
 			try
 			{
-				//For some reason, using  TextBlock and getting the AlphaMask
-				//generates a shadow with a size more smaller than the control size. 
-				if (element is TextBlock textElement)
+				var visual = ElementCompositionPreview.GetElementVisual(element);
+				bool isClipped = visual.Clip is not null;
+
+				if (!isClipped && element is TextBlock textElement)
 				{
 					return textElement.GetAlphaMask();
 				}
-				if (element is Image image)
+				if (!isClipped && element is Image image)
 				{
 					return image.GetAlphaMask();
 				}
-				if (element is Shape shape)
+				if (!isClipped && element is Shape shape)
 				{
 					return shape.GetAlphaMask();
 				}
+				if (!isClipped && element is ContentPanel contentPanel)
```

Comment:
Using the Border Shape to get the AlphaMask directly without recreate a Bitmap is massive performance boost applying shadows to Borders.